src/parser/rogue/outlaw/modules/talents/AlacrityTimesByStacks.js

```diff
@@ -0,0 +1,84 @@
+import Analyzer from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+
+/*
+  alacrityTimesByStacks() returns an array with the durations of each BS charge
+*/
+
+const MAX_ALA_STACKS = 5;
+
+class AlacrityTimesByStacks extends Analyzer {
+  alacrityStacks = [];
+  lastAlaStack = 0;
+  lastAlaUpdate = this.owner.fight.start_time;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.ALACRITY_TALENT.id);
+
+    this.alacrityStacks = Array.from({length: MAX_ALA_STACKS + 1}, x => []);
+  }
+
+  handleStacks(event, stack = null) {
```

Comment:
There is a fabricated event `changebuffstack` . It already unifies all the different buff events in to 1 that you can handle.